### DISCUSSION TITLE
Sanity check on regular_expression_gen execution time

### DISCUSSION THREAD

**Discussion by eaubin** [sgl-project/sglang#197] [Q&A] [answered]
On a v100, the regular_expression_gen from the readme takes about 2 minutes. Is this expected, or does it mean something is probably wrong with my setup?

```python
@sgl.function
def regular_expression_gen(s):
    s += "Q: What is the IP address of the Google DNS servers?\n"
    s += "A: " + sgl.gen(
        "answer",
        temperature=0,
        regex=r"((25[0-5]|2[0-4]\d|[01]?\d\d?).){3}(25[0-5]|2[0-4]\d|[01]?\d\d?)",
    )
```

**Comment by comaniac** [comment, marked as answer]
SRT takes a long time to generate FSM for the regex when seeing that for the first time. In other words, you should get an obvious speedup starting from the second request with the same regex. Also note that the generated FSM is persist on disk, so you shouldn't have to wait for another 2 minutes even you re-launch the server. On the other hand, if you change the regex or re-launch the server on a different file system (e.g., different user or docker container), then you may need to wait again.
